zephyr/esp32c3/src/common/esp_read_mac.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2021 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "esp_system.h"
+#include <rom/efuse.h>
+#include <zephyr.h>
+
+esp_err_t esp_read_mac(uint8_t *mac, esp_mac_type_t type)
```

Comment:
For both esp32 and esp32s2, this functon is currently defined under `zephyr/esp_shared/src/common/esp_system_api.c`. Should we make the same for esp32c3?

Reply:
It is a bit different in the C3, I have plans of make this function common for it as well but in a later MR :)

Reply:
Alternatively, you could consider unifying this function call in this same PR but in another commit, as done here:

https://github.com/zephyrproject-rtos/hal_espressif/pull/60/commits/9fcbf539faeee1220f57f56c82eda893f12912c6#diff-d3ff68eb35224ac623662cbaf1b15b2ca3d8131bc799c49b8c1550069ba79d3c

Reply:
I considered but it will take more time than just drop esp32c3 definitions, even the EFUSE addresses are not properly documented. I really prefer to do this in a separate PR later.